src/Cache/Engine/MemcachedEngine.php

```diff
@@ -145,7 +145,25 @@ public function init(array $config = []): bool
         }
         $this->_setOptions();
 
-        if (count($this->_Memcached->getServerList())) {
+        $serverList = $this->_Memcached->getServerList();
+        if (count($serverList)) {
+            if ($this->_config['persistent'] !== false) {
+                $actualServers = [];
+                foreach ($serverList as $server) {
+                    $actualServers[] = $server['host'] . ':' . $server['port'];
+                }
+                unset($server);
+                sort($actualServers);
+                $configuredServers = $this->_config['servers'];
+                sort($configuredServers);
+                if ($actualServers !== $configuredServers) {
+                    $message = 'Invalid cache configuration. Multiple persistent cache configurations are detected' .
+                        ' with different `servers` values. `servers` values for persistent cache configurations' .
+                        ' must be the same when using the same persistence id.';
+                    throw new InvalidArgumentException($message);
+                }
```

Comment:
```suggestion
                foreach ($serverList as $server) {
                    if (!in_array($server['host'] . ':' . $server['port'], $this->_config['server'], true)) {
                        throw new InvalidArgumentException(
                            'Invalid cache configuration. Multiple persistent cache configurations are detected' .
                            ' with different `servers` values. `servers` values for persistent cache configurations' .
                            ' must be the same when using the same persistence id.'
                        );
                    }
                }
                $actualServers = [];
                foreach ($serverList as $server) {
                    $actualServers[] = $server['host'] . ':' . $server['port'];
                }
                unset($server);
                sort($actualServers);
                $configuredServers = $this->_config['servers'];
                sort($configuredServers);
                if ($actualServers !== $configuredServers) {
                    $message = 'Invalid cache configuration. Multiple persistent cache configurations are detected' .
                        ' with different `servers` values. `servers` values for persistent cache configurations' .
                        ' must be the same when using the same persistence id.';
                    throw new InvalidArgumentException($message);
                }
```

Reply:
I don't know if that's the best exception message, but I'm not familiar with it.